setup.py

```diff
@@ -14,10 +31,18 @@
     packages=setuptools.find_packages(exclude=["tests", "tests.*"]),
     license="MIT",
     classifiers=[
+        "Programming Language :: Python :: 3.7",
+        "Programming Language :: Python :: 3.8",
+        "Programming Language :: Python :: 3.9",
+        "Programming Language :: Python :: 3.10",
         "Programming Language :: Python :: 3.11",
         "License :: OSI Approved :: MIT License",
         "Operating System :: OS Independent",
     ],
-    python_requires=">=3.11",
-    install_requires=["aiohttp", "pydantic", "loguru"]
+    python_requires=">=3.7",
```

Comment:
I am not sure about supporting python 3.7, because according to [PEP537](https://peps.python.org/pep-0537/#lifespan) that python version almost reached its end of life. Maybe it will be better to set minimal python requirement as python3.8? Also that version introduces `typing.Literal` and `typing.TypedDict` so there is no need to import mentioned types from `typing_extensions`.

Reply:
Regarding the issue with Python 3.7, I am inclined to continue supporting it for the following reasons:

1. Regarding the additional dependency, `typing_extensions`, it cannot be avoided because your type hints include features that are only available in higher versions of Python. Even if you want to retain annotations like `ParamSpec`, importing `typing_extensions` is still necessary for Python 3.10.

2. Therefore, regardless of whether Python 3.7 is nearing its end of life, `typing_extensions` is still required.

3. Based on this point, I suggest maintaining support for Python 3.7. Even if Python 3.7 has reached its end of life, it would be beneficial to continue supporting it for a certain period (e.g., three to six months) and include `DeprecationWarning` within that version. Afterward, the support can be dropped.